File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {


        int left=0;
        int n=s.size();

        unordered_set<int>st;
        int maxlen=0;
        for(int right=0;right<n;right++){

            while(st.count(s[right])){
                st.erase(s[left]);
                left++;
            }

            st.insert(s[right]);

        maxlen=max(maxlen,right-left+1);
        }
        return maxlen;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        a=set()
        left=0
        maxlen=0
        right=0
        for right in range(len(s)):
            while s[right] in a:
                a.remove(s[left])
                left=left+1
            a.add(s[right])
            maxlen=max(maxlen,right-left+1)
        return maxlen
        